src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1489,18 +1504,51 @@ export class ResultsViewPageStore {
         invoke: ()=>Promise.resolve(indexHotspotData(this.hotspotData))
     });
 
+    readonly hotspotDataForOncoprint = remoteData({
+        await:()=>[
+            this.hotspotDataPromise
+        ],
+        invoke: ()=>{
+            if (this.mutationAnnotationSettings.hotspots) {
+                return addTimeoutToPromise(
+                    this.hotspotDataPromise.result!.promise,
+                    HOTSPOTS_TIMEOUT_MS
+                );
+            } else {
+                return Promise.resolve(undefined);
+            }
+        },
+        onError:()=>{
+            alert("Hotspot data not available, or timeout encountered. Please try again in a few moments.");
```

Comment:
Should we alert inside a UI component instead?

Reply:
I feel like its okay to trigger the alert here as opposed to in the UI, it really should be triggered by the error event. @alisman what do you think?

Reply:
I don't think we should be using alert.